sui-execution/latest/sui-adapter/src/programmable_transactions/context.rs

```diff
@@ -830,9 +830,15 @@ mod checked {
             }
 
             if protocol_config.enable_coin_deny_list_v2() {
-                let (result, _num_regulated_transfers) =
-                    state_view.check_coin_deny_list(&written_objects);
-                // TODO: Charge gas based on the number of regulated transfers.
+                let DenyListResult {
+                    result,
+                    num_deny_lists_checked,
+                    num_regulated_coin_owners,
+                } = state_view.check_coin_deny_list(&written_objects);
+                gas_charger.charge_regulated_transfers(
```

Comment:
is it ok we are doing this after `refund_max_gas_budget(&mut additional_writes, gas_charger, gas_id)?;`? in line 705. It seems we do not charge after that in normal conditions?

Reply:
Let me see if I can move it. Good catch

Reply:
I don't think this is a concern, since that is just to account for the amount of gas available for the programmable transaction